src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift

```diff
@@ -96,8 +96,45 @@ final class TicketValidationCoordinator {
 
 	}
 
-	private func showSecondConsentScreen(selectedCertificate: HealthCertificate) {
-
+	private func showSecondConsentScreen(
+		selectedCertificate: HealthCertificate,
+		selectedCertifiedPerson: HealthCertifiedPerson
+	) {
+		let secondConsentViewController = SecondTicketValidationConsentViewController(
+			viewModel: SecondTicketValidationConsentViewModel(
+				serviceIdentity: ticketValidation.initializationData.serviceIdentity,
+				serviceProvider: ticketValidation.initializationData.serviceProvider,
+				healthCertificate: selectedCertificate,
+				healthCertifiedPerson: selectedCertifiedPerson,
+				onDataPrivacyTap: {
+					self.showDataPrivacy()
+				}
+			),
+			onPrimaryButtonTap: { [weak self] isLoading in
+				isLoading(true)
+				// call grantSecondConsent()
```

Comment:
You can already add the call here. That's why we have the mock, so we can build the complete flow already. The function that we need to use here is called `validate(completion:)`. And in the completion closure you can already add the switch with the error handling and call `showResultScreen(for:)` in the success case.

Reply:
Thanks, i was not aware which method should be called here. I will add them.